browser/brave_wallet/brave_wallet_provider_delegate_impl.cc

```diff
@@ -217,6 +277,86 @@ void BraveWalletProviderDelegateImpl::GetAllowedAccounts(
           selected_account));
 }
 
+void BraveWalletProviderDelegateImpl::GetAllowedAccounts(
+    mojom::CoinType type,
+    bool include_accounts_when_locked,
+    GetAllowedAccountsCallback callback) {
+  keyring_service_->GetSelectedAccount(
+      type, base::BindOnce(
+                &BraveWalletProviderDelegateImpl::ContinueGetAllowedAccounts,
+                weak_ptr_factory_.GetWeakPtr(), include_accounts_when_locked,
+                std::move(callback)));
+}
+
+void BraveWalletProviderDelegateImpl::RequestSolanaPermission(
+    RequestSolanaPermissionCallback callback) {
+  // Check if there's already a permission request in progress
+  auto* rfh = content::RenderFrameHost::FromID(host_id_);
+  if (rfh && permissions::BraveWalletPermissionContext::HasRequestsInProgress(
+                 rfh, permissions::RequestType::kBraveSolana)) {
+    std::move(callback).Run(
+        "", mojom::SolanaProviderError::kUserRejectedRequest,
+        l10n_util::GetStringUTF8(IDS_WALLET_ALREADY_IN_PROGRESS_ERROR));
+    return;
+  }
+
+  IsSelectedAccountAllowed(
+      mojom::CoinType::SOL,
+      base::BindOnce(
+          &BraveWalletProviderDelegateImpl::ContinueRequestSolanaPermission,
+          weak_ptr_factory_.GetWeakPtr(), std::move(callback)));
+}
+
+void BraveWalletProviderDelegateImpl::IsSelectedAccountAllowed(
+    mojom::CoinType type,
+    IsSelectedAccountAllowedCallback callback) {
+  keyring_service_->GetSelectedAccount(
+      type,
+      base::BindOnce(
+          &BraveWalletProviderDelegateImpl::ContinueIsSelectedAccountAllowed,
+          weak_ptr_factory_.GetWeakPtr(), std::move(callback)));
+}
+
+void BraveWalletProviderDelegateImpl::ContinueRequestSolanaPermission(
```

Comment:
as discussed in dm we should pull the provider specific logic back into the provider so we can make these methods generic and keep provider specific logic inside the provider

Reply:
fixed in 11118ed4c0feffa0e551a8e08ffb893720a73351